File: app/src/main/java/eu/toma/dev/playground/warcraft/talent/TalentService.java
package eu.toma.dev.playground.warcraft.talent;

import javax.inject.Inject;

import eu.toma.dev.playground.warcraft.api.WoWApi;
import eu.toma.dev.playground.warcraft.talent.Talents;
import eu.toma.dev.playground.warcraft.model.Talent;
import io.reactivex.Observable;

@Talents
public class TalentService
{
    private final WoWApi api;

    @Inject
    public TalentService(WoWApi api)
    {
        this.api = api;
    }

    public Observable<Talent> getTalents(){
        // TODO
        return null;
    }
}

File: app/src/main/kotlin/warcraft/talents/TalentService.kt
package warcraft.talents

import javax.inject.Inject

import io.reactivex.Observable
import warcraft.api.WoWApi
import warcraft.model.Talent

@Talents
class TalentService
@Inject
constructor(private val api: WoWApi) {

    val talents: Observable<Talent>?
    // TODO to be implemented
        get() = null
}

